### PR TITLE
Fix Properties::SortKey trailing zeros

### PR DESCRIPTION
**Pull request by Speierers**
This PR is fixing another edge case in `Properties::SortKey`: when the key's numerical suffixes compile to the same value, but have a different number of trailing zeros, the operator should fallback to the string comparison.

For instance, currently the following two keys can be used together in a `props`: `key_001` and `key_01`.

I also added a unit test that covers this edge case.

### CODE REVIEW

File: src/core/properties.cpp [hunk 1]
```diff
@@ -64,7 +64,8 @@ struct SortKey {
             long long l2 = std::strtoll(b_ptr, &b_end, 10);
             if (a_end == (a.c_str() + a.size()) &&
                 b_end == (b.c_str() + b.size()) &&
-                l1 != LLONG_MAX && l2 != LLONG_MAX)
+                l1 != LLONG_MAX && l2 != LLONG_MAX &&
+                !((l1 == l2) && (a.size() != b.size()))) // catch trailing zeros case (e.g. 001 vs 01))
```
Comment (njroussel) [inline src/core/properties.cpp:68]:
```suggestion
                !((l1 == l2) && (a.size() != b.size()))) // catch leading zeros case (e.g. 001 vs 01))
```

**Review by njroussel — APPROVED** [review]
LGTM

**Comment by njroussel** [comment]
Thank you @Speierers !
